### PR TITLE
Simplify Style/CommentedKeyword

### PR DESCRIPTION
**Pull request by buehmann**
I cleaned up this cop. All the code for heredoc detection was a leftover from a time when the cop didn't use `processed_source.comments` but looked at raw lines.

### CODE REVIEW

File: lib/rubocop/cop/style/commented_keyword.rb [hunk 1]
```diff
@@ -61,25 +47,19 @@ def investigate(processed_source)
...
+          line = line(comment)
+          KEYWORDS.any? { |word| line =~ /^\s*#{word}\s/ } &&
             ALLOWED_COMMENTS.none? { |c| line =~ /#\s*#{c}/ }
         end
 
-        def message(node)
-          line = node.source_line
-          keyword = /^\s*(\S+).*#/.match(line)[1]
+        def message(comment)
+          keyword = line(comment)[/(\S+).*#/, 1]
```
Comment (bbatsov) [inline lib/rubocop/cop/style/commented_keyword.rb]: I think using an explicit `match` is a bit more readable. I've rarely seen the index syntax in the wild and it's likely going to confuse some people.

**Comment by bbatsov** [comment]
Nicely done! Deleted code is the best code! 😉

File: lib/rubocop/cop/style/commented_keyword.rb (hunk 1 above)
Comment (buehmann) [reply lib/rubocop/cop/style/commented_keyword.rb]: Maybe we need a cop for this … 😜 Changed.

Comment (bbatsov) [reply lib/rubocop/cop/style/commented_keyword.rb]: I wouldn't mind one. :D In general I think there are very few cases where implementing `[]` results in some more readable code, as for most people it has a very strong association with some indices.

**Pull request merged**